sway/config/output.c

```diff
@@ -299,6 +318,21 @@ bool apply_output_config(struct output_config *oc, struct sway_output *output) {
 		wlr_output_set_scale(wlr_output, oc->scale);
 	}
 
+	if (oc) {
+		enum scale_filter_mode scale_filter;
+		if (oc->scale_filter == SCALE_FILTER_SMART ||
+				oc->scale_filter == SCALE_FILTER_UNKNOWN) {
```

Comment:
This feels like a hack, but without it I hit the assert in `set_scale_filter` when sway starts up because output->scale_filter is SCALE_FILTER_UNKNOWN.

I include the unkown state to make output config merging work, and I treat unknown the same as smart because smart is the default. Let me know if you've got another idea.